### PR TITLE
#160617652 implement highlighting and commenting text in articles

### PR DESCRIPTION
**Pull request by nakiwuge**
#### What does this PR do?
- Allows users to highlight and comment on text in articles
#### Description of Task to be completed?
- Ensure authenticated users are able to highlight and comment on text in article via a textbox
#### How should this be manually tested?
- Go to /view-articles, click on any article and highlight any text in the article body
#### What are the relevant pivotal tracker stories?
#160617652 https://www.pivotaltracker.com/n/projects/2198902
#### Screenshots 
screen shot 2018-11-21 at 13 13 11
screen shot 2018-11-21 at 13 13 54

### CODE REVIEW

**Comment by kimbugp** [comment]
LGTM

**Comment by LehruAngela** [comment]
@zellymiriam the comment pop up will look better if it appears in the right space of the text highlighted, and doesnt cover up the text, for a better UX. Please implement that.

File: src/assets/comment/comment.scss [hunk 1]
```diff
@@ -73,22 +73,27 @@ $border: 1px solid rgba(0, 0, 0, 0.09);
...
 .btn:active {
   outline: 0;
   box-shadow: none !important;
 }
+.comment-box {
+ width:500px
+}
+.selected-box{
+  color: grey;
+ }
```
Comment (LehruAngela) [inline src/assets/comment/comment.scss]: Add new line at end of file

File: src/__tests__/articleTests/viewArticle.test.js [hunk 2]
```diff
@@ -53,4 +53,6 @@ it("viewing article", () => {
   };
   let component = shallow(<ViewArticle {...props} />);
   component;
+  component.instance().componentDidMount();
```
Comment (LehruAngela) [inline src/__tests__/articleTests/viewArticle.test.js]: Write actual tests. When you're writing tests, you often need to check that values meet certain conditions using expect.

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/assets/comment/comment.scss (hunk 1 above)
Comment (nakiwuge) [reply src/assets/comment/comment.scss]: @LehruAngela have made this change

**Comment by nakiwuge** [comment]
@LehruAngela have made the changes you requested for

File: src/assets/comment/comment.scss (hunk 1 above)
Comment (LehruAngela) [reply src/assets/comment/comment.scss]: There seems to be no change though @zellymiriam

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
The comments need to be aligned with the articles as mentioned by the PO in the sprint call @zellymiriam Please make that update.

File: src/assets/comment/comment.scss [hunk 3]
```diff
@@ -73,22 +73,28 @@ $border: 1px solid rgba(0, 0, 0, 0.09);
...
+
```
Comment (LehruAngela) [inline src/assets/comment/comment.scss:100]: Add new line at end of file

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/assets/comment/comment.scss [hunk 4]
```diff
@@ -73,22 +73,28 @@ $border: 1px solid rgba(0, 0, 0, 0.09);
...
 .btn:active {
   outline: 0;
   box-shadow: none !important;
 }
+.comment-box {
+ width:500px
+}
+.selected-box{
+  color: grey;
+ }
```
Comment (LehruAngela) [inline src/assets/comment/comment.scss:99]:
```suggestion
}
```

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/assets/comment/comment.scss (hunk 3 above)
Comment (nakiwuge) [reply src/assets/comment/comment.scss:100]: @LehruAngela I have added the line

**Pull request merged**